### PR TITLE
enh(notion block parents): throw after 5x transient errors

### PR DESCRIPTION
**Pull request by fontanierh**
When we get rate-limited by the notion API, it's possible that we'll end up getting many transient errors in a row.
The current logic for resolving notion block parents will give up after 8 tries and return null, which means we might silently fail to get a parent if we are rate-limited

This changes the logic to throw an error (and fail the activity) in such cases. This will let Temporal retry the failing activity until we are no longer rate-limited.

@spolu:
- I kept the "max depth" at 8, but maybe that's a bit high (since the original code was doing 8 tries for both transient errors and parent layers
- I used 5 for "max_transient_errors" as it's consistent with what we use in the rest of the file, but I could also bump it a bit. WDYT ?

### CODE REVIEW

File: connectors/src/connectors/notion/lib/notion_api.ts [hunk 1]
```diff
@@ -374,15 +382,26 @@ async function getBlockParent(
           })(blockParent);
           return null;
       }
+
+      depth += 1;
+      if (depth === max_depth) {
+        // We don't want to go up more than 8 levels.
+        return null;
+      }
     } catch (e) {
```
Comment (spolu) [inline connectors/src/connectors/notion/lib/notion_api.ts:391]: If we have a series of errors but they are not isAPIResponseError then we have an infinite loop no?

Comment (fontanierh) [reply connectors/src/connectors/notion/lib/notion_api.ts:391]: Oups yes

Comment (spolu) [reply connectors/src/connectors/notion/lib/notion_api.ts:391]: Yep

Comment (fontanierh) [reply connectors/src/connectors/notion/lib/notion_api.ts:391]: Fixed

**Review by spolu — APPROVED** [review]

**Pull request merged**